rpgJavaInterpreter-core/src/test/resources/NEGATION.rpgle

```diff
@@ -0,0 +1,32 @@
+      *
+      * WORKING WITH INDICATOR 15 AND EXPRESSIONS
+      *
+
+     D NUM             S              3  0
```

Comment:
Add variable `NUMD` with scale greater than zero and test it.
Tests for both `NUM` and `NUMD` variables also when negative.

Add another test where you try to negate a variable defined as string, I want to see what happens.

Reply:
I implemented this case of test:
```
      *
      * NEGATION OF STRING
      *

     D STR             S             100          VARYING INZ('10')
     D STR_NEG         S             100          VARYING
     D RES             S             100          VARYING

     C                   EVAL      STR_NEG=-"STR"
     C                   EVAL      RES=%CHAR(STR_NEG)
     C* Excepted Exception
     C     RES           DSPLY
```
Rightly, the application explode in `RpgParserFacade.kt`

![immagine](https://github.com/smeup/jariko/assets/130480196/687d37ad-563d-49c3-82a8-e350b0a09d0c)

What is the expected result?

Reply:
You are not negating a variable defined as string but a string literal, and then:

- create another test program, for example: `NEGATIONERR.rpgle`
- create a test unit and use the test annotation for which the expected result is an error

Reply:
Sorry, typo error. Now is `C                   EVAL      STR_NEG=-STR`
The result of test is:
![immagine](https://github.com/smeup/jariko/assets/130480196/f40e4e98-8017-47f6-a67a-3ded73ac5b98)